public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx

```diff
@@ -1,75 +1,198 @@
+  options: Array<{
+    label: string;
+    value: ContactPointWithMetadata;
+    description: React.JSX.Element;
+  }>;
   onSelectContactPoint: (contactPoint?: ContactPointWithMetadata) => void;
+  refetchReceivers: () => QueryActionCreatorResult<
+    QueryDefinition<
+      string,
+      BaseQueryFn<BackendSrvRequest>,
+      'AlertmanagerChoice' | 'AlertmanagerConfiguration' | 'OnCallIntegrations' | 'OrgMigrationState',
+      AlertManagerCortexConfig,
+      'alertingApi'
+    >
+  >;
 }
-export function ContactPointSelector({ alertManager, contactPoints, onSelectContactPoint }: ContactPointSelectorProps) {
-  const styles = useStyles2(getStyles);
-  const { control, watch } = useFormContext<RuleFormValues>();
 
-  const options = contactPoints.map((receiver) => {
-    const integrations = receiver?.grafana_managed_receiver_configs;
-    const description = <ContactPointReceiverSummary receivers={integrations ?? []} />;
+export function ContactPointSelector({
+  alertManager,
+  options,
+  onSelectContactPoint,
+  refetchReceivers,
+}: ContactPointSelectorProps) {
+  const styles = useStyles2(getStyles);
+  const { control, watch, trigger } = useFormContext<RuleFormValues>();
 
-    return { label: receiver.name, value: receiver, description };
-  });
+  const contactPointInForm = watch(`contactPoints.${alertManager}.selectedContactPoint`);
 
-  const selectedContactPointWithMetadata = options.find(
-    (option) => option.value.name === watch(`contactPoints.${alertManager}.selectedContactPoint`)
-  )?.value;
+  const selectedContactPointWithMetadata = options.find((option) => option.value.name === contactPointInForm)?.value;
   const selectedContactPointSelectableValue = selectedContactPointWithMetadata
     ? { value: selectedContactPointWithMetadata, label: selectedContactPointWithMetadata.name }
     : undefined;
 
+  const LOADING_SPINNER_DURATION = 1000;
+
+  const [loadingContactPoints, setLoadingContactPoints] = useState(false);
+  // we need to keep track if the fetching takes more than 1 second, so we can show the loading spinner until the fetching is done
+  const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+  // if we have a contact point selected, check if it still exists in the event that someone has deleted it
+  const validateContactPoint = useCallback(() => {
+    if (contactPointInForm) {
+      trigger(`contactPoints.${alertManager}.selectedContactPoint`, { shouldFocus: true });
+    }
+  }, [alertManager, contactPointInForm, trigger]);
+
+  const onClickRefresh = () => {
+    setLoadingContactPoints(true);
+    Promise.all([refetchReceivers(), sleep(LOADING_SPINNER_DURATION)]).finally(() => {
+      setLoadingContactPoints(false);
+      validateContactPoint();
+    });
+  };
+
+  // validate the contact point and check if it still exists when mounting the component
+  useEffect(() => {
+    validateContactPoint();
+  }, [validateContactPoint]);
```

Comment:
This `useEffect` will be invoked on each `contactPointInForm` change. Is it desired?

Reply:
yes!!! this is the exact desired behaviour